lib/templatesimplifier.cpp

```diff
@@ -695,6 +708,75 @@ void TemplateSimplifier::getTemplateInstantiations()
                     qualification += (qualification.empty() ? "" : " :: ") + tok->str();
                 tok = tok->tokAt(2);
             }
+
+            // look for function instantiation with type deduction
+            // fixme: only single argument functions supported
```

Comment:
Is it already worthwhile to add a TODO testcase for functions with more arguments?

Reply:
I can do that in another patch. I'll see if the clang test suite has an example.  I just wanted to get this minimal implementation integrated while I work on fixing bugs found while trying to fix #8972.

Reply:
I added a minimal TODO test case for the FIXME.